src/commands/utils/clusterfilter.ts

```diff
@@ -0,0 +1,95 @@
+import { ManagedCluster } from "@azure/arm-containerservice";
+import { getAksClient } from "../utils/arm";
+import { getReadySessionProvider } from "../../auth/azureAuth";
+import { getFilteredClusters, setFilteredClusters } from "./config";
+import { parseResource, parseSubId } from "../../azure-api-utils";
+
+export default async function aksClusterFilter(_context: IActionContext, target: unknown): Promise<void> {
+    const cloudExplorer = await k8s.extension.cloudExplorer.v1;
+
+    const subscriptionNode = getAksClusterSubscriptionNode(target, cloudExplorer);
+    if (failed(subscriptionNode)) {
+        vscode.window.showErrorMessage(subscriptionNode.error);
+        return;
+    }
+
+    const extension = getExtension();
+    if (failed(extension)) {
+        vscode.window.showErrorMessage(extension.error);
+        return;
+    }
+
+    const sessionProvider = await getReadySessionProvider();
+    if (failed(sessionProvider)) {
+        vscode.window.showErrorMessage(sessionProvider.error);
+        return;
+    }
+
+    const containerServiceClient = getAksClient(sessionProvider.result, subscriptionNode.result.subscriptionId);
+    const clusterList: ManagedCluster[] = [];
+
+    await longRunning(`Getting AKS Cluster list for ${subscriptionNode.result.name}`, async () => {
+        const iterator = containerServiceClient.managedClusters.list();
+        for await (const clusters of iterator.byPage()) {
+            const validClusters = clusters.filter((c) => c.id && c.name);
+            clusterList.push(
+                ...validClusters.map((c) => ({ label: c.name!, name: c.name!, id: c.id!, location: c.location! })),
+            );
+        }
+    });
+
+    const filteredClusters = await getUniqueClusters();
+
+    const quickPickItems: ClusterQuickPickItem[] = clusterList.map((cluster) => {
+        return {
+            label: cluster.name || "",
+            description: cluster.name,
+            picked: filteredClusters.some((filtered) => filtered.clusterName === parseResource(cluster.id!).name), // filtered.subscriptionId === parseSubId(cluster.id!).subId),
+            Cluster: {
+                clusterName: cluster.name || "",
+                subscriptionId: parseSubId(cluster.id!).subId || "",
+            },
+        };
+    });
+
+    // show picked items at the top
+    quickPickItems.sort((itemA, itemB) => (itemA.picked === itemB.picked ? 0 : itemA.picked ? -1 : 1));
+
+    const selectedItems = await vscode.window.showQuickPick(quickPickItems, {
+        canPickMany: true,
+        placeHolder: "Select Cluster",
+    });
+
+    if (!selectedItems) {
+        return;
+    }
+
+    // Set Cluster Instance
+    const newFilteredClusters = [
+        ...selectedItems.map((item) => item.Cluster), // Retain filters in any other tenants.
+    ];
+
+    await setFilteredClusters(newFilteredClusters);
+}
+
+async function getUniqueClusters() {
+    const filteredClusters = getFilteredClusters();
```

Comment:
Initially this function confused me that container service client was listing duplicates but then realized this is list of clusters that are already selected and stored in the user settings config. can we rename the function ex: getSelectedClustersFromConfig ? or give a description to the function method with description of what it does Please ?

Reply:
ok, lets chat : `getFileterestCluster` means all the cluster which are filtered? `getSelectedClustersFromConfig` is more kind of developer level lingo, is what I think, I am open to thoughts. Lets chat. Thanks.

Reply:
Agreed for consistency purposes. we can refactor the method names later. Thank you.